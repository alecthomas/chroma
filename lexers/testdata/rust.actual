#[derive(Debug)]
struct Rectangle {
    width: u32,
    height: u32,
}

fn main() {
    let rect1 = Rectangle { width: 30, height: 50 };

    // Some comment
    println!("rect1 is {:?}", rect1);

    let r#type = "valid";
}