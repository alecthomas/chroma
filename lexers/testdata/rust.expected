[
  {"type":"CommentPreproc","value":"#[derive(Debug)]"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Keyword","value":"struct"},
  {"type":"Text","value":" "},
  {"type":"NameClass","value":"Rectangle"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"Name","value":"width"},
  {"type":"Text","value":": "},
  {"type":"KeywordType","value":"u32"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"Name","value":"height"},
  {"type":"Text","value":": "},
  {"type":"KeywordType","value":"u32"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Keyword","value":"fn"},
  {"type":"Text","value":" "},
  {"type":"NameFunction","value":"main"},
  {"type":"Punctuation","value":"()"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"KeywordDeclaration","value":"let"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"rect1"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"Rectangle"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"width"},
  {"type":"Text","value":": "},
  {"type":"LiteralNumberInteger","value":"30"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"height"},
  {"type":"Text","value":": "},
  {"type":"LiteralNumberInteger","value":"50"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"};"},
  {"type":"TextWhitespace","value":"\n\n    "},
  {"type":"CommentSingle","value":"// Some comment\n"},
  {"type":"TextWhitespace","value":"    "},
  {"type":"Name","value":"println"},
  {"type":"Operator","value":"!"},
  {"type":"Punctuation","value":"("},
  {"type":"LiteralString","value":"\"rect1 is {:?}\""},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"rect1"},
  {"type":"Punctuation","value":");"},
  {"type":"TextWhitespace","value":"\n\n    "},
  {"type":"KeywordDeclaration","value":"let"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"r#type"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralString","value":"\"valid\""},
  {"type":"Punctuation","value":";"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":"\n"}
]
